consensus-types/types/payload_requests.go

```diff
@@ -50,46 +52,56 @@ type newPayloadRequest struct {
 	// versionedHashes is the versioned hashes of the execution payload.
 	versionedHashes []common.ExecutionHash
 	// parentBeaconBlockRoot is the root of the parent beacon block.
-	parentBeaconBlockRoot *common.Root
+	parentBeaconBlockRoot common.Root
 	// ExecutionRequests is introduced in Pectra. It is only non-nil after Pectra.
 	executionRequests []EncodedExecutionRequest
+	// ParentProposerPubkey is introduced in Pectra1. It is only non-nil after Pectra1.
+	parentProposerPubkey *crypto.BLSPubkey
```

Comment:
aiming to standardize this pattern from geth throughout:

For these types which are "optional" (i.e. not in every fork version) we keep them as pointers or nillable. For types like parentBeaconBlockRoot, this is required in every fork version we support so choosing to keep it not nil.